packages/react-components/react-carousel-preview/library/src/components/useEmblaCarousel.ts

```diff
@@ -16,13 +17,60 @@ const DEFAULT_EMBLA_OPTIONS: EmblaOptionsType = {
 
 export const EMBLA_VISIBILITY_EVENT = 'embla:visibilitychange';
 
-export function useEmblaCarousel({ align, direction, loop }: Pick<EmblaOptionsType, 'align' | 'direction' | 'loop'>) {
-  const emblaOptions = React.useRef<EmblaOptionsType>({ align, direction, loop });
+export function useEmblaCarousel(
+  options: Pick<EmblaOptionsType, 'align' | 'direction' | 'loop' | 'slidesToScroll'> & {
+    defaultActiveIndex: number | undefined;
+    activeIndex: number | undefined;
+  },
+) {
+  const { align, direction, loop, slidesToScroll } = options;
+  const [activeIndex, setActiveIndex] = useControllableState({
+    defaultState: options.defaultActiveIndex,
+    state: options.activeIndex,
+    initialState: 0,
+  });
+
+  const emblaOptions = React.useRef<EmblaOptionsType>({
+    align,
+    direction,
+    loop,
+    slidesToScroll,
+    startIndex: activeIndex,
+  });
   const emblaApi = React.useRef<EmblaCarouselType | null>(null);
 
-  const ref = React.useMemo(() => {
+  // Listeners contains callbacks for UI elements that may require state update based on embla changes
+  const listeners = React.useRef(new Set<(data: CarouselUpdateData) => void>());
```

Comment:
nit: instead of keeping track of listeners, you can use a simple `EventTarget`.

```tsx
  const [eventTarget] = React.useState(() => new EventTarget());
  const subscribeForValues = React.useCallback((listener: (e: CustomEvent<CarouselUpdateData>) => void) => {
    eventTarget.addEventListener('carousel-update', listener as EventListener);

    return () => {
      eventTarget.removeEventListener('carousel-update', listener as EventListener);
    };
  }, []);
```

Reply:
Going to explore this as part of our stable updates - going to make this public shortly so users can begin testing implementation.